bot/seasons/halloween/monsterbio.py

```diff
@@ -0,0 +1,52 @@
+import json
+import logging
+import random
+from pathlib import Path
+
+import discord
+from discord.ext import commands
+
+from bot.constants import Colours
+
+log = logging.getLogger(__name__)
+
+with open(Path("bot/resources/halloween/monster.json"), "r", encoding="utf8") as f:
+    TEXT_OPTIONS = json.load(f)  # Data for a mad-lib style generation of text
+
+
+class MonsterBio(commands.Cog):
+    """A cog that generates a spooky monster biography."""
+
+    def __init__(self, bot: commands.Bot):
+        self.bot = bot
+
+    def generate_name(self, length: int) -> str:
+        """Generates a name (for either monster species or monster name)."""
+        return "".join([random.choice(TEXT_OPTIONS["monster_type"][i]) for i in range(length)])
+
+    @commands.command(brief="Sends your monster bio!")
+    async def monsterbio(self, ctx: commands.Context) -> None:
+        """Sends a description of a monster."""
+        random.seed(ctx.message.author.id)
+        name = self.generate_name(random.randint(2, len(TEXT_OPTIONS["monster_type"])))
+        species = self.generate_name(random.randint(2, len(TEXT_OPTIONS["monster_type"])))
+        biography_text = random.choice(TEXT_OPTIONS["biography_text"])
+        words = {"monster_name": name, "monster_species": species}
+        for key, value in biography_text.items():
+            if key == "text":
+                continue
+            if value > 1:
+                words[key] = random.sample(TEXT_OPTIONS[key], value)
+            else:
+                words[key] = random.choice(TEXT_OPTIONS[key])
```

Comment:
This future-proofed code has a bug in it, `random.sample` returns a list.

I suggest changing this entire if block to
```py
options = random.sample(TEXT_OPTIONS[key], value)
words[key] = ' '.join(options)
```
or similar